external/java/src/main/java/ch/epfl/dedis/lib/omniledger/contracts/DarcInstance.java

```diff
@@ -0,0 +1,159 @@
+     * @param pos     position of the instruction in the ClientTransaction
+     * @param len     total number of instructions in the ClientTransaction
+     * @return Instruction to be sent to omniledger
+     * @throws CothorityCryptoException
+     */
+    public Instruction evolveDarcInstruction(Darc newDarc, Signer owner, int pos, int len) throws CothorityCryptoException {
+        newDarc.increaseVersion();
+        newDarc.setPrevId(darc);
+        if (!newDarc.getBaseId().equals(darc.getBaseId()) ||
+                newDarc.getVersion() != darc.getVersion() + 1) {
+            logger.info("{}, {}", darc.getVersion(), newDarc.getVersion());
+            throw new CothorityCryptoException("not correct darc to evolve");
+        }
+        logger.info("mydarc: {}", darc.getBaseId());
+        logger.info("newDarc: {} / {}", newDarc.getId(), newDarc.getBaseId());
+        Invoke inv = new Invoke("evolve", "darc", newDarc.toProto().toByteArray());
+        Instruction inst = new Instruction(new ObjectID(darc.getBaseId(), ObjectID.zero()),
+                ObjectID.zero(), pos, len, inv);
+        try {
+            Request r = new Request(darc.getBaseId(), "invoke:evolve", inst.hash(),
+                    Arrays.asList(owner.getIdentity()), null);
+            logger.info("Signing: {}", DatatypeConverter.printHexBinary(r.hash()));
+            Signature sign = new Signature(owner.sign(r.hash()), owner.getIdentity());
+            inst.setSignatures(Arrays.asList(sign));
+        } catch (Signer.SignRequestRejectedException e) {
+            throw new CothorityCryptoException(e.getMessage());
+        }
+        return inst;
+    }
+
+    public TransactionId evolveDarc(Darc newDarc, Signer owner) throws CothorityException {
+        Instruction inst = evolveDarcInstruction(newDarc, owner, 0, 1);
+        ClientTransaction ct = new ClientTransaction(Arrays.asList(inst));
+        ol.sendTransaction(ct);
+        return new TransactionId(darc.getBaseId(), new SubId(inst.hash()));
+    }
+
+    /**
+     * Asks omniledger to evolve the darc and waits until the new darc has
+     * been stored in the global state.
+     * TODO: check if there has been an error in the transaction!
+     *
+     * @param newDarc is the new darc to replace the old one
+     * @param owner   is the owner that can sign to evolve the darc
+     * @throws CothorityException
+     */
+    public void evolveDarcAndWait(Darc newDarc, Signer owner) throws CothorityException {
+        TransactionId tid = evolveDarc(newDarc, owner);
+        while (true) {
+            Proof p = ol.getProof(instance.getId());
+            Instance inst = new Instance(p);
+            try {
+                darc = new Darc(inst.getData());
+                if (darc.getVersion() == newDarc.getVersion()) {
+                    break;
+                }
+                Thread.sleep(ol.getConfig().getBlockInterval().getNano() / 1_000_000);
```

Comment:
getBlockInterval() returns a Duration, and Thread.sleep wants milis, so use toMillis: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toMillis--

Reply:
thanks - I did a quick google search and only found this solution... Fixed it.